lib/crash_reporter.js

```diff
@@ -0,0 +1,136 @@
+};
+
+CrashReporter.getInstance = function() {
+  if (crashReporter) {
+    return crashReporter;
+  } else {
+    crashReporter = new CrashReporter();
+    return crashReporter;
+  }
+};
+
+CrashReporter.prototype.turnOff = function() {
+  return this.preferences
+    .write(CRASH_REPORTER_PREFERENCE, 'off')
+    .catch(function(error) {
+      // do nothing
+      // do not crash the crash reporter :)
+      logs.err('Error turning off crash reporter preferences', error);
+    });
+};
+
+CrashReporter.prototype.turnOn = function() {
+  return this.preferences
+    .write(CRASH_REPORTER_PREFERENCE, 'on')
+    .catch(function(error) {
+      // do nothing
+      // do not crash the crash reporter :)
+      logs.err('Error turning on crash reporter preferences', error);
+    });
+};
+
+CrashReporter.prototype.submit = function(report) {
+  var that = this;
+  that.preferences.read(CRASH_REPORTER_PREFERENCE, 'on')
+    .then(function(value) {
+      if (value === 'on') {
+        var labels = [
+          // tessel cli labels
+          packageJson.name,
+          `CLI version: ${packageJson.version}`,
+          // node version
+          `Node version: ${process.version}`,
+          // os version
+          `OS platform: ${os.platform()}`,
+          `OS release: ${os.release()}`
+        ];
+
+        that.submitReport(labels.join(','), report)
+          .then(function() {
+            logs.info('Done !');
+          })
+          .catch(function(error) {
+            logs.err('Error submitting crash report', error);
+          });
+      }
+    })
+    .catch(function(error) {
+      // do nothing
+      // do not crash the crash reporter :)
+      logs.err('Error submitting crash report', error);
+    });
+};
+
+CrashReporter.prototype.submitReport = function(labels, report) {
+  return new Promise(function(resolve, reject) {
+    request.post({
+      'url': SUBMIT_CRASH_URL,
+      'form': {
+        'crash': report,
+        'labels': labels,
+        'f': 'json'
+      }
+    }, function(error, httpResponse, body) {
+      try {
+        if (error) {
+          reject(error);
+        } else {
+          var json = JSON.parse(body);
+          if (json['error']) {
+            reject(json['error']);
+          } else {
+            resolve();
+          }
+        }
+      } catch (exception) {
+        reject(exception);
+      }
+    });
+  });
+};
+
+var onError = function(error) {
+  return CrashReporter.getInstance().submit(error.stack);
+};
+
+var CrashReportsHelper = function() {
+
+};
+
+CrashReportsHelper.turnOff = function() {
+  return CrashReporter.getInstance().turnOff();
+};
+
+CrashReportsHelper.turnOn = function() {
+  return CrashReporter.getInstance().turnOn();
+};
+
+CrashReportsHelper.testSubmit = function() {
+  return Promise.reject(new Error('Testing the crash reporter'));
+};
+
+process.on('unhandledRejection', onError);
+process.on('uncaughtException', onError);
+
+module.exports = CrashReportsHelper;
```

Comment:
I think you can get rid of `CrashReportsHelper` entirely now. Just move `testSubmit` to `CrashReporter` and then change this line to `module.export = CrashReporter;`